### PR TITLE
Save multiple payment methods

### PR DESCRIPTION
**Pull request by malteo**

### CODE REVIEW

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/MyUserDTO.java [hunk 1]
```diff
@@ -136,6 +136,9 @@
     @ApiModelProperty( value = "If the user has a valid payment method available", readOnly = true )
     private boolean paymentDataSaved;
 
+    @ApiModelProperty( value = "The user saved payment method labels", readOnly = true )
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/MyUserDTO.java]: MINOR Remove this use of "readOnly"; it is deprecated. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/MyUserDTO.java [hunk 2]
```diff
@@ -17,6 +17,7 @@
 import javax.validation.constraints.Pattern;
 import javax.validation.constraints.Size;
 import java.util.Date;
+import java.util.Map;
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/MyUserDTO.java]: MINOR Remove this unused import 'java.util.Map'. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/UserPaymentMethodDTO.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.cloudesire.platform.apiclient.dto.model.dto;
+
+public class UserPaymentMethodDTO extends BaseEntityDTO
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/UserPaymentMethodDTO.java]: MINOR Override the "equals" method in this class. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS2160)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java [hunk 4]
```diff
@@ -175,9 +184,17 @@
     @DELETE( "user/{id}" )
     Call<Void> delete( @Path( "id" ) int id );
 
+    /**
+     * @deprecated look for {@link PaymentMethodApi#delete(int)}
+     */
+    @Deprecated
     @DELETE( "user/{id}/payment" )
     Call<Void> deletePaymentDataForAdmin( @Path( "id" ) int id );
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java [hunk 5]
```diff
@@ -145,9 +146,17 @@
...
+    @Deprecated
     @POST( "user/payment/token" )
     Call<Void> saveTokenData( @Body TokenDataDTO data );
 
+    /**
+     * @deprecated look for {@link PaymentMethodApi#create(UserPaymentMethodDTO)}
+     */
+    @Deprecated
     @POST( "user/payment/method" )
     Call<Void> savePaymentMethod( @Body PaymentMethodDTO data );
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java [hunk 6]
```diff
@@ -175,9 +184,17 @@
...
+    @Deprecated
     @DELETE( "user/{id}/payment" )
     Call<Void> deletePaymentDataForAdmin( @Path( "id" ) int id );
 
+    /**
+     * @deprecated look for {@link PaymentMethodApi#delete(int)}
+     */
+    @Deprecated
     @DELETE( "user/payment" )
     Call<Void> deletePaymentData();
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java [hunk 7]
```diff
@@ -145,9 +146,17 @@
     @POST( "user/payment/sepa" )
     Call<Void> saveSepaData( @Body SepaDataDTO data );
 
+    /**
+     * @deprecated look for {@link PaymentMethodApi#create(UserPaymentMethodDTO)}
+     */
+    @Deprecated
     @POST( "user/payment/token" )
     Call<Void> saveTokenData( @Body TokenDataDTO data );
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 4 issues
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/ApiVersion.java [hunk 8]
```diff
@@ -22,7 +23,7 @@
     public static final long V20180101 = 20180101;
     public static final long NO_VERSION = 0;
 
-    public static final Long LATEST_API_VERSION = V20200108;
+    public static final Long LATEST_API_VERSION = V20200114;
```
Comment (gionn) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/ApiVersion.java]: rename before merge

**Comment by malteo** [comment]
rename PaymentMethodDTO and UserPaymentMethodDTO before merge

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java [hunk 9]
```diff
@@ -132,24 +133,32 @@
...
-    Call<Void> saveTokenData( @Body TokenDataDTO data );
+    Call<Void> saveTokenData( @Body StripeTokenDTO data );
 
+    /**
+     * @deprecated look for {@link PaymentMethodApi#create(PaymentMethodDTO)}
+     */
+    @Deprecated
     @POST( "user/payment/method" )
-    Call<Void> savePaymentMethod( @Body PaymentMethodDTO data );
+    Call<Void> savePaymentMethod( @Body StripePaymentMethodDTO data );
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java [hunk 10]
```diff
@@ -132,24 +133,32 @@
...
     @POST( "user/payment/sepa" )
     Call<Void> saveSepaData( @Body SepaDataDTO data );
 
+    /**
+     * @deprecated look for {@link PaymentMethodApi#create(PaymentMethodDTO)}
+     */
+    @Deprecated
     @POST( "user/payment/token" )
-    Call<Void> saveTokenData( @Body TokenDataDTO data );
+    Call<Void> saveTokenData( @Body StripeTokenDTO data );
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java [hunk 11]
```diff
@@ -132,24 +133,32 @@
...
     @POST( "user/payment/card" )
     Call<Void> saveCardData( @Body CardDataDTO data );
 
     /**
-     * @deprecated look for {@link #saveTokenData(TokenDataDTO)}
+     * @deprecated look for {@link #saveTokenData(StripeTokenDTO)}
      */
     @Deprecated
     @POST( "user/payment/sepa" )
     Call<Void> saveSepaData( @Body SepaDataDTO data );
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java [hunk 12]
```diff
@@ -132,24 +133,32 @@
     Call<Void> requestVerification();
 
     /**
-     * @deprecated look for {@link #savePaymentMethod(PaymentMethodDTO)}
+     * @deprecated look for {@link #savePaymentMethod(StripePaymentMethodDTO)}
      */
     @Deprecated
     @POST( "user/payment/card" )
     Call<Void> saveCardData( @Body CardDataDTO data );
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/SepaDataDTO.java [hunk 13]
```diff
@@ -4,7 +4,7 @@
 import org.hibernate.validator.constraints.NotEmpty;
 
 /**
- * @deprecated by {@link TokenDataDTO}
+ * @deprecated by {@link StripeTokenDTO}
  */
 @Deprecated
 public class SepaDataDTO extends PaymentDataDTO
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/SepaDataDTO.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 15 issues
* ![INFO][INFO] 15 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [CardDataDTO.java#L18](https://github.com/ClouDesire/java-api-client/blob/f22ad2aa464cbf801aa2f8748e382f5f471563ab/cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/CardDataDTO.java#L18): Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [CloudesireClient.java#L246](https://github.com/ClouDesire/java-api-client/blob/f22ad2aa464cbf801aa2f8748e382f5f471563ab/cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/CloudesireClient.java#L246): Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [CloudesireClient.java#L325](https://github.com/ClouDesire/java-api-client/blob/f22ad2aa464cbf801aa2f8748e382f5f471563ab/cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/CloudesireClient.java#L325): Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [CloudesireClient.java#L334](https://github.com/ClouDesire/java-api-client/blob/f22ad2aa464cbf801aa2f8748e382f5f471563ab/cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/CloudesireClient.java#L334): Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [UserApi.java#L53](https://github.com/ClouDesire/java-api-client/blob/f22ad2aa464cbf801aa2f8748e382f5f471563ab/cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java#L53): Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [UserApi.java#L93](https://github.com/ClouDesire/java-api-client/blob/f22ad2aa464cbf801aa2f8748e382f5f471563ab/cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java#L93): Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [UserApi.java#L174](https://github.com/ClouDesire/java-api-client/blob/f22ad2aa464cbf801aa2f8748e382f5f471563ab/cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java#L174): Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [UserApi.java#L181](https://github.com/ClouDesire/java-api-client/blob/f22ad2aa464cbf801aa2f8748e382f5f471563ab/cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserApi.java#L181): Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Pull request merged**